cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -198,10 +198,13 @@ private void executeExistingRolesRequest(Role role, HttpRequest request) throws
     }
   }
 
-  private HttpResponse executePrivilegeRequest(EntityId entityId, HttpRequest request) throws FeatureDisabledException,
-    UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
+  private HttpResponse executePrivilegeRequest(EntityId entityId, Principal principal, HttpRequest request) throws
+    FeatureDisabledException, UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
     HttpResponse httpResponse = doExecuteRequest(request, HttpURLConnection.HTTP_NOT_FOUND);
     if (HttpURLConnection.HTTP_NOT_FOUND == httpResponse.getResponseCode()) {
+      if (!listAllRoles().contains(new Role(principal.getName()))) {
```

Comment:
The check is done in the security extension code [https://github.com/caskdata/cdap-security-extn/blob/develop/cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/SentryAuthorizer.java#L100]. So in this case, I should change the security extension code and check if role exists there otherwise return an HTTP 500 Error to the client?